### PR TITLE
1 0 recording refactor

### PR DESCRIPTION
**Pull request by kalturaguy**

### CODE REVIEW

File: recordingUploader/Tasks/UploadTask.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                                recorded_obj.replacementStatus)
+                self.backend_client.cancel_replace(upload_session.partner_id, self.recorded_id)
+            self.backend_client.set_recorded_content_remote(upload_session, str(float(self.duration)/1000))
+            os.rename(self.output_file_path, self.output_file_path + '.done')
+        else:
+            raise Exception("Failed to upload file, "+str(len(failed_jobs))+" chunks from "+str(chunks_to_upload)+ " where failed:"
+                            + upload_session_json)
+
+    def append_recording_handler(self):
+        partner_id = self.backend_client.get_live_entry(self.entry_id).partnerId
```
Comment (kalturaguy) [inline recordingUploader/Tasks/UploadTask.py]: add +        self.check_stamp()

File: recordingUploader/Tasks/UploadTask.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+    def upload_file(self, file_name):
+
+        self.check_stamp()
+        threadWorkers = ThreadWorkers()
+        file_size = os.path.getsize(file_name)
+        chunks_to_upload = self.get_chunks_to_upload(file_size)
+        infile = io.open(file_name, 'rb')
```
Comment (kalturaguy) [inline recordingUploader/Tasks/UploadTask.py]: do with

File: packager/config/nginx.conf.template [hunk 3]
```diff
@@ -106,6 +106,15 @@ http {
...
         }
 
+        vod_force_continuous_timestamps on;
+        location ~ ^/recording/hls/e/(?<entryId>[^/]+)/ {
+            vod_force_continuous_timestamps on;
+            alias @CONTENT_DIR@/$entryId/playlist.json;
+            vod hls;
+        }
+
+        include nginx.conf.redirect.rules;
```
Comment (kalturaguy) [inline packager/config/nginx.conf.template]: please remove

File: lib/entry/StateManager.js [hunk 4]
```diff
@@ -58,6 +58,9 @@ module.exports = function(liveEntry) {
                 logger.info(stateMachineMessagePattern, event.toUpperCase(), from, to);
                 return (liveEntry.entryObject.liveStatus === 1) ? Q.resolve() : sync.exec(function() {
                     return backendClient.registerEntryInDatabase(liveEntry.entryObject, KalturaLiveStatus.BROADCASTING, event)
+                        .then( (entryObj)=> {
```
Comment (kalturaguy) [inline lib/entry/StateManager.js]: please check with yossi

File: lib/playlistGenerator/PlaylistGenerator.js [hunk 5]
```diff
@@ -256,7 +286,7 @@ var doPersist =  function() {
         that.playlistImp.validate();
         var jsonStr = JSON.stringify(that.playlistImp);
         if (false && playlistUtils.playlistConfig.debug) {
-            Playlist.prototype.serializeFrom(jsonStr, that.loggerInfo,function(newPlaylist){
+            Playlist.prototype.serializeFrom(that.playlistType, jsonStr, that.loggerInfo,function(newPlaylist){
```
Comment (igorshevach) [inline lib/playlistGenerator/PlaylistGenerator.js]: playlistType already present in jsonStr , so it's not necessary

File: packager/config/nginx.conf.template [hunk 6]
```diff
@@ -105,6 +105,13 @@ http {
                 vod mss;
             }
         }
+        location ~ ^/recording/hls/e/(?<entryId>[^/]+)/ {
+            vod_force_continuous_timestamps on;
+            alias @CONTENT_DIR@/$entryId/playlist.json;
+            vod hls;
+        }
+
+        include nginx.conf.redirect.rules;
```
Comment (kalturaguy) [inline packager/config/nginx.conf.template]: remove +        include nginx.conf.redirect.rules;

**Pull request merged**